packages/xo-web/src/xo-app/xoa/support/index.js

```diff
@@ -22,35 +26,83 @@ const Support = decorate([
       checkXoa: async () => ({ stdoutCheckXoa: await checkXoa() }),
     },
+    computed: {
+      stdoutSupportTunnel: (_, { tunnelState }) =>
+        tunnelState === undefined
+          ? undefined
+          : { __html: marked(tunnelState.stdout) },
+    },
   }),
   injectState,
-  ({ effects, state: { stdoutCheckXoa } }) => (
+  ({
+    effects,
+    state: { stdoutCheckXoa, stdoutSupportTunnel },
+    tunnelState = { open: false },
+  }) => (
     <Container>
+      {COMMUNITY && (
+        <Row className='mb-2'>
+          <Col>
+            <span className='text-info'>{_('supportCommunity')}</span>
+          </Col>
+        </Row>
+      )}
       <Row>
         <Col mediumSize={6}>
           <Card>
             <CardHeader>{_('xoaCheck')}</CardHeader>
-            {COMMUNITY ? (
-              <CardBlock>
-                <span className='text-info'>{_('checkXoaCommunity')}</span>
-              </CardBlock>
-            ) : (
-              <CardBlock>
-                <ActionButton
-                  btnStyle='success'
-                  handler={effects.checkXoa}
-                  icon='diagnosis'
-                >
-                  {_('checkXoa')}
-                </ActionButton>
-                <hr />
-                <pre
-                  dangerouslySetInnerHTML={{
-                    __html: ansiUp.ansi_to_html(stdoutCheckXoa),
-                  }}
-                />
-              </CardBlock>
-            )}
+            <CardBlock>
+              <ActionButton
+                btnStyle='success'
+                disabled={COMMUNITY}
+                handler={effects.checkXoa}
+                icon='diagnosis'
+              >
+                {_('checkXoa')}
+              </ActionButton>
+              <hr />
+              <pre
+                dangerouslySetInnerHTML={{
+                  __html: ansiUp.ansi_to_html(stdoutCheckXoa),
+                }}
+              />
+            </CardBlock>
+          </Card>
+        </Col>
+        <Col mediumSize={6}>
+          <Card>
+            <CardHeader>{_('supportTunnel')}</CardHeader>
+            <CardBlock>
+              <Row>
+                <Col mediumSize={6}>
+                  {tunnelState.open ? (
+                    <ActionButton
+                      btnStyle='primary'
+                      disabled={COMMUNITY}
+                      handler={closeTunnel}
+                      icon='remove'
+                    >
+                      {_('closeTunnel')}
+                    </ActionButton>
+                  ) : (
+                    <ActionButton
+                      btnStyle='success'
+                      disabled={COMMUNITY}
+                      handler={openTunnel}
+                      icon='open-tunnel'
+                    >
+                      {_('openTunnel')}
+                    </ActionButton>
+                  )}
+                </Col>
+              </Row>
+              <hr />
+              {tunnelState.open ? (
```

Comment:
We want to show the error if the tunnel couldn't be opened, for example.

Reply:
@pdonias 
The `open` props mean that the opening process is started.
The `stdout` contains `errors`.

If the tunnel couldn't be opened, the `error` will be displayed.

Reply:
What if the process starts, fails to connect or whatever, and ends with an error. Then we'd see the tunnel as "closed" (or at least we should), but we'd still want to see the output, right?

Reply:
You're right.